app/scripts/models/notifications.js

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * The notifier broadcasts messages across multiple channels (iframe, tabs, browsers, etc).
+ */
+
+'use strict';
+
+define([
+  'backbone',
+  'underscore',
+  'lib/promise',
+  'lib/constants',
+  'lib/channels/web'
+], function (Backbone, _, p, Constants, WebChannel) {
+
+  var Notifier = Backbone.Model.extend({
+    defaults: {
+      events: {
+        PROFILE_IMAGE_CHANGE: 'profile_image_change'
+      }
+    },
+    initialize: function (options) {
+      options = options || {};
+      this._tabChannel = options.tabChannel;
+      this._iframeChannel = options.iframeChannel;
+      this._webChannel = options.webChannel;
+
+      if (! this._webChannel) {
+        this._webChannel = new WebChannel(Constants.PROFILE_WEBCHANNEL_ID);
+        this._webChannel.init(options);
+      }
+
+      this._listen();
+    },
+
+    broadcast: function (event, data) {
+      this._webChannel.send(event, data);
+      this._tabChannel.send(event, data);
+      if (this._iframeChannel) {
```

Comment:
Maybe an option is to put all of the channels into an array, then fire off the event using an [].forEach instead of keeping track of each manually and using guard clauses. 

If the iframeChannel is created in app-start, is this guard needed?